sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AQEShuffleReadExec.scala

```diff
@@ -112,6 +128,14 @@ case class AQEShuffleReadExec private(
     partitionSpecs.exists(_.isInstanceOf[PartialMapperPartitionSpec]) ||
       partitionSpecs.exists(_.isInstanceOf[CoalescedMapperPartitionSpec])
 
+  def isCoalescedRead: Boolean = {
+    // We need to make sure there is no repeated partition spec, which is the repeated side of the
+    // skew join shuffle reader and is not a coalesced reader.
+    partitionSpecs.forall(_.isInstanceOf[CoalescedPartitionSpec]) && partitionSpecs.collect {
+      case CoalescedPartitionSpec(start, end, _) => start -> end
+    }.distinct.length == partitionSpecs.length
```

Comment:
not sure, but would for loop would be more efficient here? just check "last end" == "current start".